src/CPUSolver.cpp

```diff
@@ -323,6 +398,91 @@ void CPUSolver::computeFSRSources() {
   }
 }
 
+/**
+ * @brief Computes the total fission source in each FSR.
+ * @details This method is a helper routine for the openmoc.krylov submodule.
+ */
+void CPUSolver::computeFSRFissionSources() {
+
+  #pragma omp parallel default(none)
+  {
+    int tid;
+    Material* material;
+    FP_PRECISION* sigma_t;
+    FP_PRECISION fiss_mat;
+    FP_PRECISION fission_source;
+    FP_PRECISION* fission_sources = new FP_PRECISION[_num_groups];
+
+    /* Compute the total source for each FSR */
+    #pragma omp for schedule(guided)
+    for (int r=0; r < _num_FSRs; r++) {
+
+      tid = omp_get_thread_num();
+      material = _FSR_materials[r];
+      sigma_t = material->getSigmaT();
+
+      /* Compute scatter + fission source for group G */
+      for (int G=0; G < _num_groups; G++) {
```

Comment:
Personally, I'm uneasy with the use of ``G`` as a variable name for the scatter source. I would much prefer something like ``g`` and ``g_prime`` or if we want brevity ``g`` and ``gp`` might be alright.